src/core/security/client_auth_filter.c

```diff
@@ -199,8 +200,28 @@ static void auth_start_transport_op(grpc_call_element *elem,
   channel_data *chand = elem->channel_data;
   grpc_linked_mdelem *l;
   size_t i;
+  grpc_client_security_context* sec_ctx = NULL;
 
-  /* TODO(jboeuf): write the call auth context. */
+  if (calld->security_context_set == 0) {
+    calld->security_context_set = 1;
+    GPR_ASSERT(op->context);
+    if (op->context[GRPC_CONTEXT_SECURITY].value == NULL) {
+      op->context[GRPC_CONTEXT_SECURITY].value =
+          grpc_client_security_context_create();
+      op->context[GRPC_CONTEXT_SECURITY].destroy =
+          grpc_client_security_context_destroy;
+    }
+    sec_ctx = op->context[GRPC_CONTEXT_SECURITY].value;
+    if (sec_ctx->auth_context == NULL) {
+      sec_ctx->auth_context =
+          GRPC_AUTH_CONTEXT_REF(chand->security_connector->base.auth_context,
+                                "client_auth_filter");
+    } else {
+      sec_ctx->auth_context->chained =
```

Comment:
I am attaching an auth_context of the security_connector to an existing auth_context of a call, and I am reffing the former auth_context, since it will be unreffed when the client_security_context is destroyed.
Am I correct in this?